src/utility/bucketizeColumns.js

```diff
@@ -0,0 +1,47 @@
+const buckets = 40;
+
+export default (columns, xVals, histogramName) => {
+  if (columns.length <= buckets) {
+    return {
+      [histogramName]: columns,
+      x: xVals,
+    };
+  }
+  const max = Math.max(...xVals);
+  const bucketSize = ~~(max / buckets);
+  const newXVals = [];
+  let i = 0;
+  for (i; i < buckets; i++) {
+    newXVals.push(bucketSize * i);
+  }
+  i = 0;
+  const newColumns = newXVals.map((val) => {
+    const newObj = { win: 0, games: 0 };
+    let enteredLoop = false;
+    while (i < xVals.length && xVals[i] <= val) {
```

Comment:
I'd like to leave it as is for now, I can't think of a way of doing this right now without increasing runtime by a large amount (since this only does O(columns). If something comes to me that would be a more functional way without O(n^2) then I will reimplement.